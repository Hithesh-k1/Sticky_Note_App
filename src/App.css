@import url("https://fonts.googleapis.com/css2?family=Covered+By+Your+Grace&family=Oregano&display=swap");

/* input[type="text"],
input[type="text"]:focus {
  background: transparent;
  border: none;
  outline-width: 0;
} */

.container {
  display: flex;
  justify-content: space-evenly;
  align-items: flex-start;
}

.parent {
  display: grid;
  grid-template-columns: 1fr 0.2fr;
  grid-template-rows: 0.2fr 1fr 0.3fr;
  grid-column-gap: 7px;
  grid-row-gap: 11px;
  height: 100vh;
}

.header {
  grid-area: 1 / 1 / 2 / 3;
  /* font-family: 'Covered By Your Grace', cursive; */
font-family: 'Oregano', cursive;
}
.content {
  grid-area: 3 / 2 / 4 / 3;
}
.add-notes {
  grid-area: 2 / 1 / 4 / 3;
}
