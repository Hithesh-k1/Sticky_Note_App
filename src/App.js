import React, { useState, Fragment } from "react";
import AddNote from "./components/AddNote";
import EditNote from "./components/EditNote";
import Notes from "./components/Notes";
import "./App.css";

const App = () => {
  // Data
  const usersData = [
    { id: 1, title: "Todo 1", notes: "build CI/CD pipeline using jenkins" },
    { id: 2, title: "Todo 2", notes: "Integrate Jenkins with Sonarqube " },
    { id: 3, title: "Todo 3", notes: "Deploy to AWS S3" },
  ];

  const initialFormState = { id: null, title: "", notes: "" };

  const [notes, setNotes] = useState(usersData);
  const [currentNote, setCurrentNote] = useState(initialFormState);
  const [editing, setEditing] = useState(false);

  // CRUD operations
  const addNote = (note) => {
    note.id = notes.length + 1;
    setNotes([...notes, note]);
  };

  const deleteNote = (id) => {
    setEditing(false);
    setNotes(notes.filter((note) => note.id !== id));
  };

  const updateNote = (id, updatedUser) => {
    setEditing(false);

    setNotes(notes.map((note) => (note.id === id ? updatedUser : note)));
  };

  const editRow = (note) => {
    setEditing(true);

    setCurrentNote({ id: note.id, title: note.title, notes: note.notes });
  };

  console.log(editing,"--------------------")
  return (

    <div className='parent' >
      <div>
        <h1 className='header'>Sticky-notes</h1>
      </div>
        <div className='content' >
          {editing ? (
            <Fragment>
              <h2>Edit note</h2>

              <EditNote
                editing={editing}
                setEditing={setEditing}
                currentNote={currentNote}
                updateNote={updateNote}
              />
            </Fragment>
          ) : (
            <Fragment>
              <h2>Add Note</h2>

              <AddNote addNote={addNote} />
            </Fragment>
          )}
        </div>
        <div className='add-notes'>

        <div className="container">
          <Notes notes={notes} editRow={editRow} deleteNote={deleteNote} />
        </div>
        </div>
    </div>
  );
};

export default App;
