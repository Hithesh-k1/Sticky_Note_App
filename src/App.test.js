import React from "react";
import { shallow } from "enzyme";
import App from "./App";

let wrapped = shallow(<App />);

describe("App component", () => {
  it("should match snapshot of App Component", () => {
    expect(wrapped).toMatchSnapshot();
  });
});
