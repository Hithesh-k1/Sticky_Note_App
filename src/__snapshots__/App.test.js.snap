// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot of App Component 1`] = `
<div
  className="parent"
>
  <div>
    <h1
      className="header"
    >
      Sticky-notes
    </h1>
  </div>
  <div
    className="content"
  >
    <h2>
      Add Note
    </h2>
    <AddNote
      addNote={[Function]}
    />
  </div>
  <div
    className="add-notes"
  >
    <div
      className="container"
    >
      <Notes
        deleteNote={[Function]}
        editRow={[Function]}
        notes={
          Array [
            Object {
              "id": 1,
              "notes": "build CI/CD pipeline using jenkins",
              "title": "Todo 1",
            },
            Object {
              "id": 2,
              "notes": "Integrate Jenkins with Sonarqube ",
              "title": "Todo 2",
            },
            Object {
              "id": 3,
              "notes": "Deploy to AWS S3",
              "title": "Todo 3",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
